### PR TITLE
Convert all 'nil =='style to '== nil' style

### PR DESCRIPTION
**Pull request by chaodaiG**
Currently there is a mixture of `nil == err` and `err == nil` styles, and the former one was mainly introduced by me, the primary purpose was to avoid accidental variable assignment like
```
if err = nil {
}
```
Which is always true in some other languages. This is not the case in go as it would complain `syntax error: assignment err = nil used as value`. This change convert all `nil == err` usages to `err == nil` to ensure consistency

### CODE REVIEW

File: shared/gcs/gcs.go [hunk 1]
```diff
@@ -48,13 +48,13 @@ func Authenticate(ctx context.Context, serviceAccount string) error {
 func Exists(ctx context.Context, bucketName, storagePath string) bool {
 	// Check if this is a file
 	handle := createStorageObject(bucketName, storagePath)
-	if _, err := handle.Attrs(ctx); nil == err {
+	if _, err := handle.Attrs(ctx); err == nil {
```
Comment (srinivashegde86) [inline shared/gcs/gcs.go]: these can be combined
```
if _, err := handle.Attrs(); err != nil {
  _, err = getObjectsIter().Next()
  return err == nil
}
return true
```

File: shared/gcs/gcs.go [hunk 2]
```diff
@@ -95,7 +95,7 @@ func Download(ctx context.Context, bucketName, srcPath, dstPath string) error {
 		return err
 	}
 	defer src.Close()
-	if _, err = io.Copy(dst, src); nil != err {
+	if _, err = io.Copy(dst, src); err != nil {
```
Comment (srinivashegde86) [inline shared/gcs/gcs.go]: no need of if
```
_, err = io.Copy()
return err
```

File: shared/gcs/gcs.go [hunk 3]
```diff
@@ -104,11 +104,11 @@ func Download(ctx context.Context, bucketName, srcPath, dstPath string) error {
 // Upload file to gcs
 func Upload(ctx context.Context, bucketName, dstPath, srcPath string) error {
 	src, err := os.Open(srcPath)
-	if nil != err {
+	if err != nil {
 		return err
 	}
 	dst := createStorageObject(bucketName, dstPath).NewWriter(ctx)
-	if _, err = io.Copy(dst, src); nil != err {
+	if _, err = io.Copy(dst, src); err != nil {
```
Comment (srinivashegde86) [inline shared/gcs/gcs.go]: ditto. No need of if
```
defer dst.Close()
_, err = io.Copy()
return err
```

File: shared/prow/prow.go [hunk 4]
```diff
@@ -195,7 +195,7 @@ func (j *Job) GetFinishedBuilds() []Build {
 	var finishedBuilds []Build
 	builds := j.GetBuilds()
 	for _, build := range builds {
-		if true == build.IsFinished() {
+		if build.IsFinished() == true {
```
Comment (srinivashegde86) [inline shared/prow/prow.go]:
```suggestion
		if build.IsFinished() {
```

File: shared/prow/prow.go [hunk 5]
```diff
@@ -328,10 +328,10 @@ func getBuildIDFromBuildPath(buildPath string) (int, error) {
 // v must be an arbitrary struct, slice, or string.
 func unmarshalJSONFile(storagePath string, v interface{}) error {
 	contents, err := gcs.Read(ctx, BucketName, storagePath)
-	if nil != err {
+	if err != nil {
 		return err
 	}
-	if err := json.Unmarshal(contents, v); nil != err {
+	if err := json.Unmarshal(contents, v); err != nil {
```
Comment (srinivashegde86) [inline shared/prow/prow.go]: ditto. no need of if
```
return json.Unmarshal()
```

File: shared/gcs/gcs.go (hunk 1 above)
Comment (chaodaiG) [reply shared/gcs/gcs.go]: I would prefer current logic: return true if it's a file, else check if it's a folder.

File: shared/gcs/gcs.go (hunk 2 above)
Comment (chaodaiG) [reply shared/gcs/gcs.go]: agree. Done

File: tools/flaky-test-reporter/result.go [hunk 6]
```diff
@@ -219,9 +219,9 @@ func (rd *RepoData) getResultSliceForTest(testName string) []junit.TestStatusEnu
 	ts := rd.TestStats[testName]
 	for i, buildID := range rd.BuildIDs {
 		switch {
-		case true == intSliceContains(ts.Failed, buildID):
+		case intSliceContains(ts.Failed, buildID) == true:
```
Comment (srinivashegde86) [inline tools/flaky-test-reporter/result.go]:
```suggestion
		case intSliceContains(ts.Failed, buildID):
```

File: tools/flaky-test-reporter/result.go [hunk 7]
```diff
@@ -219,9 +219,9 @@ func (rd *RepoData) getResultSliceForTest(testName string) []junit.TestStatusEnu
...
 			res[i] = junit.Failed
-		case true == intSliceContains(ts.Passed, buildID):
+		case intSliceContains(ts.Passed, buildID) == true:
```
Comment (srinivashegde86) [inline tools/flaky-test-reporter/result.go]:
```suggestion
		case intSliceContains(ts.Passed, buildID):
```

File: tools/prow-auto-bumper/main.go [hunk 8]
```diff
@@ -34,12 +34,12 @@ func main() {
 	dryrun := flag.Bool("dry-run", false, "dry run switch")
 	flag.Parse()
 
-	if nil != dryrun && true == *dryrun {
+	if dryrun != nil && *dryrun == true {
```
Comment (srinivashegde86) [inline tools/prow-auto-bumper/main.go]:
```suggestion
	if dryrun != nil && *dryrun {
```

File: shared/gcs/gcs.go (hunk 3 above)
Comment (chaodaiG) [reply shared/gcs/gcs.go]: done

File: shared/prow/prow.go (hunk 5 above)
Comment (chaodaiG) [reply shared/prow/prow.go]: done

**Pull request merged**